src/plugins_types.h

```diff
@@ -213,13 +213,14 @@ struct lysc_type_leafref;
  * @param[in] vecode Validity error code in case of LY_EVALID error code.
  * @param[in] path Path to the node causing the error.
  * @param[in] apptag Error-app-tag value.
- * @param[in] err_msg error formating message.
+ * @param[in] err_format Format string (same like at printf) or string literal.
  * @return The given @p ecode value if the @p err is successfully created. The structure can be freed using ::ly_err_free()
  * or passed back from callback into libyang.
  * @return LY_EMEM If there is not enough memory for allocating error record, the @p err is not touched in that case.
  * @return LY_SUCCESS if @p ecode is LY_SUCCESS, the @p err is not touched in this case.
  */
-LY_ERR ly_err_new(struct ly_err_item **err, LY_ERR ecode, LY_VECODE vecode, char *path, char *apptag, const char *err_msg, ...);
+LY_ERR ly_err_new(struct ly_err_item **err, LY_ERR ecode, LY_VECODE vecode, char *path, char *apptag, const char *err_format, ...)
+__attribute__((format (printf, 6, 7)));
```

Comment:
Make the attribute conditional, look into `compat/compat.h.in` and do it the same way as `__unused__` or `__packed__`.